apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/EndpointJsonComparatorBase.java

```diff
@@ -0,0 +1,163 @@
+package gov.cms.bfd.server.war;
+
+import static io.restassured.RestAssured.given;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.utils.AssertUtils;
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Holds shared functionality between endpoint comparator tests in v1/v2. */
+public abstract class EndpointJsonComparatorBase extends ServerRequiredTest {
+
+  /** Test to use for an ignored field. */
+  protected static final String IGNORED_FIELD_TEXT = "IGNORED_FIELD";
+
+  /**
+   * Generates current endpoint response files, comparing them to the corresponding approved
+   * responses.
+   *
+   * <p>If you've recently made mapping changes, and this test fails, you may need to re-generate
+   * the expected response files by building the project with the maven parameter
+   * -DgenerateTestData=true which will run the test that generates new response files. This should
+   * only be done if you're sure your mapping changes are correct, as these files are used for
+   * regression testing future changes. Once generated, these newly minted files should be checked
+   * in with your mapping changes.
+   *
+   * @param endpointId the endpoint id
+   * @param endpointOperation the endpoint operation
+   */
+  @ParameterizedTest(name = "endpointId = {0}")
+  @MethodSource("data")
+  public void verifyCorrectEndpointResponse(String endpointId, Supplier<String> endpointOperation) {
+    String endpointResponse = endpointOperation.get();
+    assertJsonDiffIsEmpty(endpointId, endpointResponse);
+  }
+
+  /**
+   * Compares the approved and current responses for an endpoint.
+   *
+   * @param endpointId the name of the operation being tested, used to determine which files to
+   *     compare
+   * @param endpointResponse the endpoint response
+   */
+  public void assertJsonDiffIsEmpty(String endpointId, String endpointResponse) {
+    String expectedJson =
+        ServerTestUtils.readFile(
+            ServerTestUtils.generateEndpointJsonFileName(getExpectedJsonResponseDir(), endpointId));
```

Comment:
What is this doing?  It appears to create a string and do nothing with it.  If `getExpectedJsonResponseDir()` has a side effect it would be better to not give it a name that looks like a getter.

Reply:
I renamed the method to generatePathForEndpointJsonFile which should hopefully be a little clearer.